solution.rb

```diff
@@ -0,0 +1,38 @@
+require 'nokogiri'
+require 'open-uri'
+
+class ShakespeareAnalyzer
+    attr_reader :character_lines
+
+    def initialize(url)
+        @url = url
+        @character_lines = Hash.new(0)
+    end
+
+    def analyze
+        character_lines.clear if not character_lines.empty?
```

Comment:
There are other ways to do this like:
```suggestion
        character_lines.clear if character_lines.any?
```
Or
```suggestion
        character_lines.clear unless character_lines.empty?
```

It's not too common to see `if not`